source/speech/__init__.py

```diff
@@ -230,10 +230,64 @@ def speakSpelling(
 	speak(seq, priority=priority)
 
 
-# C901 'getSpellingSpeech' is too complex
-# Note: when working on getSpellingSpeech, look for opportunities to simplify
-# and move logic out into smaller helper functions.
-def getSpellingSpeech(  # noqa: C901
+def getSpellingSpeechAddCharMode(
+		seq: Generator[SequenceItemT, None, None],
+) -> Generator[SequenceItemT, None, None]:
+	"""Inserts CharacterMode commands in a speech sequence generator to ensure any single character
+	is spelt by the synthesizer.
+	@param seq: The speech sequence to be spelt.
+	"""
+	synth = getSynth()
+	synthConfig = config.conf["speech"][synth.name]
+	charMode = False
+	for item in seq:
+		if isinstance(item, str):
+			if len(item) == 1 and synthConfig["useSpellingFunctionality"]:
+				if not charMode:
+					yield CharacterModeCommand(True)
+					charMode = True
+			elif charMode:
+				yield CharacterModeCommand(False)
+				charMode = False
+		yield item
+
+
+def getSpellingCharWithCaseInfo(

```

Comment:
Please make this a private method for now:
```suggestion
def _getSpellingCharWithCaseInfo(

```